install.sh

```diff
@@ -0,0 +1,171 @@
+}
+
+delete_pref() {
+  echo "resetting $(echo $1 | cut -d '"' -f 2) to default";
+  sed -i "/$1/d" "$PROFILE_ROOTDIR/user.js"
+
+  if $2; then
+    sed -i "/$1/d" "$PROFILE_ROOTDIR/prefs.js"
+  fi;
+}
+
+ask_question() {
+  local defaultAns="$2"
+  
+  read -e -p "$1" in
+  defaultAns="${in:-$defaultAns}";
+  defaultAns="$(echo $defaultAns | tr '[:upper:]' '[:lower:]')"
+  [[ $defaultAns =~ ^(yes|y)$ ]];
+}
+
+#####################
+# PRE-INSTALL PHASE #
+#####################
+
+firefox_proc=$(pgrep firefox);
+if [ ! -z $firefox_proc ]; then
+  echo "Before (un)installing, please make sure firefox is not running."
+  echo "Otherwise, changes cannot be made to user.js"
+  exit 0;
+fi
+
+# Prompting for correct install directory
+echo "Please enter the path to your firefox profile directory."
+echo "This can be found by opening about:support in firefox and looking for the Profile root directory."
+echo "Press CTRL+C to abort installation now."
+
+echo -e "${CYAN}Automatically detected: $PROFILE_ROOTDIR${NC}"
+read -e -p "Press ENTER to use this directory, or type in a new one: " newdir
+PROFILE_ROOTDIR="${newdir:-$PROFILE_ROOTDIR}"
+
+if [ ! -d "$PROFILE_ROOTDIR" ]; then
+  echo "ERROR: firefox profile directory could not be found"
+
+  while [ ! -d "$PROFILE_ROOTDIR" ]; do
+    read -p "Enter active root directory found in \"about:profiles\" here: " PROFILE_ROOTDIR;
+
+    if [ ! -d "$PROFILE_ROOTDIR" ]; then
+      echo "invalid directory: specified location does not exist. Try again..."
+    fi
+  done;
+fi
+
+# Check if issued `./installer.sh uninstall`
+if [[ $1 == "uninstall" ]]; then
+  echo -e "${YELLOW}NOTE: This is the final opportunity to abort uninstallation by pressing Ctrl+C${NC}";
+  if ask_question "Delete all files in $PROFILE_ROOTDIR/chrome? [Y/n]: " "y"; then
+    rm -rf $PROFILE_ROOTDIR/chrome;
+  fi;
+
+  reset_all= ask_question "Reset settings in about:config to default? (a backup of user.js and prefs.js will be created if yes) [y/N]: " "n";
+  if $reset_all; then
+    cp "$PROFILE_ROOTDIR/prefs.js" "$PROFILE_ROOTDIR/prefs.js.bak";
+    cp "$PROFILE_ROOTDIR/user.js" "$PROFILE_ROOTDIR/user.js.bak";
+  fi;
+
+  # Download changed user.js from online
+  FETCHED_PREFS=();
+  if ping -c 1 raw.githubusercontent.com &> /dev/null; then
+    while read l;
+      do FETCHED_PREFS+=("$l");
+    done < <(curl -s "https://raw.githubusercontent.com/bmFtZQ/edge-frfox/main/user.js" | grep -E "user_pref\(\"([a-zA-Z0-9.-]+)\",\s*(true|false)\);")
+  else
+    echo -e "${YELLOW}WARNING: Failed to retrieve online user.js prefs. Continuing...${NC}"
+  fi;
+
+  for ((i = 0; i < ${#FETCHED_PREFS[@]}; i++)); do
+    delete_pref "${FETCHED_PREFS[i]}" $reset_all;
+  done;
+
+  ans="y"
+  echo -e "${CYAN}NOTE: enter 'a' to answer 'yes' to all questions.${NC}";
+  for ((i = 0; i < ${#OPTIONALS[@]}; i+=2)); do
+    setting="user_pref(\"${OPTIONALS[i]}\", ${OPTIONALS[i+1]});";
+
+    if [[ ! ${ans,,} =~ ^(all|a)$ ]]; then
```

Comment:
`ask_question` returns either true or false for 'yes' or 'no' answers respectively, in order to adapt this code to work with ask_question, I'd have to remove the possibility of answering 'all'. Doing it any other way kind of makes the existence of `ask_question` pointless, since it wouldn't really shorten the code. Perhaps I should rename the function to `ask_binary_question` though, since that might be more descriptive.

Reply:
The name of the function is probably fine. Not using the `ask_question` function would just mean that the `tr` command will need to be called again (if you want to accept both upper and lower case inputs).

Maybe that could be turned into a function if the conversion is needed enough times?

Something like this:

```sh
to_lowercase() {
  echo $1 | tr "[:upper:]" "[:lower:]"
}
```

Then it can be called like this:
```sh
$(to_lowercase $input)
```

Reply:
> Not using the ask_question function would just mean that the tr command will need to be called again (if you want to accept both upper and lower case inputs).

That's already been done, check latest changes. I'll change it into a function now too, why not.